### PR TITLE
smbios: Use SMBIOS type 2 product to check OEM IPMI commands

### PR DESCRIPTION
**Pull request by johnnylinwiwynn**
Defined in DSP0134 7.3, length of the structure is at least 08h.
Without this fix dmidecode -t 2 might not display baseboard information
correctly.

Signed-off-by: Johnny Lin <johnny_lin@wiwynn.com>

### CODE REVIEW

**Comment by johnnylinwiwynn** [comment]
@pmazzini @morganjangwiwynn @JingleHsuWiwynn @dhendrix @jonzhang-fb

**Comment by johnnylinwiwynn** [comment]
I don't think this check failure is caused by my commits?
u-root.u-root — #20200518.3 failed

File: pkg/smbios/type2_baseboard_information.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func ParseBaseboardInfo(t *Table) (*BaseboardInfo, error) {
 	if t.Type != TableTypeBaseboardInfo {
 		return nil, fmt.Errorf("invalid table type %d", t.Type)
 	}
-	if t.Len() < 0xf {
+	if t.Len() < 0x8 {
```
Comment (rminnich) [inline pkg/smbios/type2_baseboard_information.go:36]: what is this number?

Comment (johnnylinwiwynn) [reply pkg/smbios/type2_baseboard_information.go:36]: It's the minimum length of type 2 structure,
defined in DSP0134 7.3, length of the structure is at least 08h. The original code would return early if the length is smaller than 0xf which should be incorrect.

Comment (rminnich) [reply pkg/smbios/type2_baseboard_information.go:36]: can you put a comment to that effect here? People need to understand why there is an 8.

Comment (johnnylinwiwynn) [reply pkg/smbios/type2_baseboard_information.go:37]: Comment added.

Comment (johnnylinwiwynn) [reply pkg/smbios/type2_baseboard_information.go:37]: Hi @insomniacslk @pmazzini, appreciate any review comment.
Thanks.

**Comment by johnnylinwiwynn** [comment]
Hi @insomniacslk @pmazzini, appreciate any review comment.
Thanks.

**Pull request merged**